### PR TITLE
Adding windows forms for PLH import dialogs

### PR DESCRIPTION
**Pull request by lilyclements**
@africanmathsinitiative/developers can someone check this over quickly to make sure the correct alterations are made (and that I have not accidentally changed anything major! I was getting an error on my computer which I believe I fixed, but would like that to be made sure!).

Once checked, can this be merged so that we can get working on the import dialogs (issues #7828 #7829)

### CODE REVIEW

**Review by N-thony — DISMISSED** [review]
@lloyddewit over to you.

File: instat/frmMain.Designer.vb [hunk 1]
```diff
@@ -5091,18 +5102,39 @@ Partial Class frmMain
         Me.mnuLogFile.Text = "Log Window..."
         Me.mnuLogFile.ToolTipText = "Log Window"
         '
+        'ToolStripMenuItem2
+        '
+        Me.ToolStripMenuItem2.Name = "ToolStripMenuItem2"
+        Me.ToolStripMenuItem2.Size = New System.Drawing.Size(351, 34)
+        Me.ToolStripMenuItem2.Text = "Convert..."
```
Comment (lloyddewit) [inline instat/frmMain.Designer.vb:5109]: there was already a 'Convert' menu option. Do we need this?

File: instat/frmMain.Designer.vb [hunk 2]
```diff
@@ -5091,18 +5102,39 @@ Partial Class frmMain
...
+        'ToolStripMenuItem3
+        '
+        Me.ToolStripMenuItem3.Enabled = False
+        Me.ToolStripMenuItem3.Name = "ToolStripMenuItem3"
+        Me.ToolStripMenuItem3.Size = New System.Drawing.Size(351, 34)
+        Me.ToolStripMenuItem3.Text = "Import From Postgres..."
+        '
+        'ToolStripMenuItem4
+        '
+        Me.ToolStripMenuItem4.Enabled = False
```
Comment (lloyddewit) [inline instat/frmMain.Designer.vb:5120]: Greyed out menu options make things more confusing for the user. Better to make enabled true and visible false?

File: instat/frmMain.Designer.vb [hunk 3]
```diff
@@ -5091,18 +5102,39 @@ Partial Class frmMain
...
         '
+        'ToolStripMenuItem2
+        '
+        Me.ToolStripMenuItem2.Name = "ToolStripMenuItem2"
+        Me.ToolStripMenuItem2.Size = New System.Drawing.Size(351, 34)
+        Me.ToolStripMenuItem2.Text = "Convert..."
+        '
+        'ToolStripMenuItem3
+        '
+        Me.ToolStripMenuItem3.Enabled = False
```
Comment (lloyddewit) [inline instat/frmMain.Designer.vb:5113]: Greyed out menu options make things more confusing for the user. Better to make enabled true and visible false?

File: instat/sdgImportFromRapidPro.vb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+﻿Public Class sdgImportFromRapidPro
```
Comment (lloyddewit) [inline instat/sdgImportFromRapidPro.vb:1]: add standard class header comment?

File: instat/dlgImportFromPostgres.vb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+﻿Public Class dlgImportFromPostgres
```
Comment (lloyddewit) [inline instat/dlgImportFromPostgres.vb:1]: add standard class header comment?

File: instat/dlgImportFromRapidPro.vb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+﻿Public Class dlgImportFromRapidPro
```
Comment (lloyddewit) [inline instat/dlgImportFromRapidPro.vb:1]: add standard class header comment?

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@lilyclements Thanks for this. I had a few suggestions

**Comment by lilyclements** [comment]
@lloyddewit thank you for the review - I have made the amendments

**Comment by lilyclements** [comment]
I'm seeing errors that say "dlgImportFromPostgres is not declared" - I'm very confused how this happened! @lloyddewit do you have any idea how to fix this?

File: instat/instat.vbproj [hunk 7]
```diff
@@ -11,7 +11,7 @@
     <AssemblyName>instat</AssemblyName>
     <FileAlignment>512</FileAlignment>
     <MyType>WindowsForms</MyType>
-    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
```
Comment (lilyclements) [inline instat/instat.vbproj:14]: @lloyddewit is this a problem on your version?

**Comment by lilyclements** [comment]
@lloyddewit have fixed it - sorry for the minor panic! One comment above that I want to be double checked before proceeding!

File: instat/instat.vbproj (hunk 7 above)
Comment (lilyclements) [reply instat/instat.vbproj:14]: Let me check something first, sorry

**Comment by lloyddewit** [comment]
> I'm seeing errors that say "dlgImportFromPostgres is not declared" - I'm very confused how this happened! @lloyddewit do you have any idea how to fix this?

This is because the new dialog boxes have not been added to the project in the standard way, so the compiler cannot find the new 'dlgImport...' classes. If we look at the Visual Studio Solution Explorer, then we can see that the new dialogs look different to the others (the files are in the directory and managed by Git but they are not part of the project).

image

Also, in the PR's list of changed files, we can also see that the 'instat.vbproj' files has not been updated (it should have been). 

I suggest you delete the 2 empty dialogs and add them again in the standard way. there should be instructions [here](https://github.com/africanmathsinitiative/R-Instat/wiki/Developer-Tutorial) (see 'your first dialog' section). @N-thony or one of the other developers should also be able to help.

Hope this helps.

**Pull request closed without merging**